src/gridded/indexing.jl

```diff
@@ -96,3 +118,39 @@ function getindex_return_type(::Type{GriddedInterpolation{T,N,TCoefs,IT,K}}, arg
     end
     Tret
 end
+
+Base.@propagate_inbounds function searchsortedfirst_exp_left(v, xx, lo, hi)
+    for i in 0:4
+        ind = lo + i
+        ind > hi && return ind
+        xx <= v[ind] && return ind
+    end
+    n = 3
+    tn2 = 2^n
+    tn2m1 = 2^(n-1)
+    ind = lo + tn2
+    while ind <= hi
+        xx <= v[ind] && return searchsortedfirst(v, xx, lo + tn2 - tn2m1, ind, Base.Order.Forward)
+        tn2 *= 2
+        tn2m1 *= 2
+        ind = lo + tn2
+    end
+    return searchsortedfirst(v, xx, lo + tn2 - tn2m1, hi, Base.Order.Forward)
+end
+
+function searchsortedfirst_vec(v::AbstractVector, x::AbstractVector)
+    issorted(x) || return searchsortedfirst.(Ref(v), x)
+    out = zeros(Int, length(x))
+    firstlo = 1
```

Comment:
Not sure what the `firstlo` and `firsthi` are doing. Could these just be `lo` and `hi`?